zero_bin/leader/src/main.rs

```diff
@@ -53,22 +54,25 @@ async fn main() -> Result<()> {
     }
 
     let args = cli::Cli::parse();
-    if let paladin::config::Runtime::InMemory = args.paladin.runtime {
-        // If running in emulation mode, we'll need to initialize the prover
-        // state here.
-        args.prover_state_config
-            .into_prover_state_manager()
-            .initialize()?;
-    }
 
     let runtime = Runtime::from_config(&args.paladin, register()).await?;
 
-    let cli_prover_config = args.prover_config;
+    let prover_config: ProverConfig = args.prover_config.into();
+
+    // If not in test_only mode and running in emulation mode, we'll need to
+    // initialize the prover state here.
+    if !prover_config.test_only {
+        if let paladin::config::Runtime::InMemory = args.paladin.runtime {
+            args.prover_state_config
+                .into_prover_state_manager()
+                .initialize()?;
```

Comment:
This isn't inlined with #288:
The goal of the issue was to not have to do _any_ kind of circuit preinitialization, if we are in test-only mode, for the sole reason that we do not use any circuit in this mode. As such, we should be able to override any ENV_VAR argument and just look in the circuit directory for any cached prover_state, loading one if we find any.

Reply:
This is particularly useful for testing / debugging something on the Kernel side, now that we use the kernel hash as a versioning tool for the cached circuit validity. We don't want to have to regenerate every single time the circuits when touching the kernel and testing.